docs/howto/manage-metrics.md

```diff
@@ -0,0 +1,167 @@
+
+The charm retrieves the ID of the secret that stores the username and password from the configuration option.
+
+In the handler for the `config-changed` event, we'll retrieve the contents of the secret and create a Pebble identity. We'll also handle the `secret-changed` event, in case the charm user changes the contents of the secret.
+
+After retrieving the contents of the secret, we'll use the [`replace_identities`](ops.pebble.Client.replace_identities) method to create a "basic" type identity in Pebble:
+
+```python
+from passlib.hash import sha512_crypt
+
+class MyCharm(ops.CharmBase):
+  ...
+
+  def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
+    # The user must have:
+    # - Created a secret with keys 'username' and 'password'
+    # - Stored the secret ID in the 'metrics-secret-id' configuration option
+    secret_id = str(self.config["metrics-secret-id"])
+    secret = self.model.get_secret(id=secret_id)
+    content = secret.get_content()
+    username, password = content["username"], content["password"]
+    self._replace_identities(username, password)
+
+  def _on_secret_changed(self, event: ops.SecretChangedEvent) -> None:
+    if event.secret.id == self.config['metrics-secret-id']:
+      content = event.secret.peek_content()
+      username, password = content["username"], content["password"]
+      self._replace_identities(username, password)
+
+  def _replace_identities(self, username: str, password: str) -> None:
+    identities = {
+      username: ops.pebble.Identity(
+        access="metrics", basic=ops.pebble.BasicIdentity(password=sha512_crypt.hash(password))
+      ),
+    }
+    self.container.pebble.replace_identities(identities)
+    logger.debug("New metrics username: %s", username)
+
+  ...
+```
+
+The password of the Pebble identity is stored as a hash, which we generate using `sha512_crypt.hash()` from `passlib.hash`.
+
+When Pebble receives a request to access the metrics endpoint, Pebble will verify that the basic authentication credentials in the request match the identity's username and password.
+
+> See more:
+> - {ref}`make-your-charm-configurable`
+> - {ref}`manage-configurations`
+> - {ref}`manage-secrets`
+
+## Deploy the charm and grant access to the user secret
+
+We'll use a configuration file to set `metrics-secret-id` to the secret ID.
+
+First, create a configuration file named `metrics-config.yaml`:
+
+```yaml
+metrics-charm:
+  metrics-secret-id: <secret-id-here>
+```
+
+Then, when deploying the charm, use the `--config` option to pass the configuration file:
+
+```bash
+juju deploy <charm-name> --config metrics-config.yaml
+```
+
+After deploying the charm, grant access to the user secret:
+
+```bash
+juju grant-secret metrics-user-password <charm-name>
+```
+
+> See more:
+> - {external+juju:ref}`Juju | config <command-juju-config>`
+> - {external+juju:ref}`Juju | deploy <command-juju-deploy>`
+> - {external+juju:ref}`Juju | grant-secret <command-juju-grant-secret>`
```

Comment:
I think config and deploy are such basic commands they don't need to be linked, and maybe we can link grant-secret in the text above: `... [grant access](...) to the user secret`.